### PR TITLE
OpenCoarrays: update to v2.2.0

### PR DESCRIPTION
**Pull request by bcfriesen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/opencoarrays/package.py [hunk 1]
```diff
@@ -48,6 +49,9 @@ class Opencoarrays(CMakePackage):
                     'MinSizeRel', 'CodeCoverage'))
 
     depends_on('mpi')
+    # This patch removes a bunch of checks for the version of MPI available on
+    # the system. They make the Crays hang.
+    patch('CMakeLists.patch')
```
Comment (mamelara) [inline var/spack/repos/builtin/packages/opencoarrays/package.py]: if this patch is only meant for Crays, then you may want to add `patch('CMakeLists.patch', when='platform=cray')`. I'm sure when this gets upstreamed they will ask for that.

**Review by mamelara — CHANGES_REQUESTED** [review]

**Comment by bcfriesen** [comment]
Oops, I guess when I did `git push -f` it messed up your comment. Anyway, I added `when='platform=cray'` as you requested.

**Review by mamelara — APPROVED** [review]

**Comment by becker33** [comment]
@bcfriesen @mamelara any chance you could make a PR for this update against the mainline Spack repo as well?

**Comment by mamelara** [comment]
done: https://github.com/spack/spack/pull/9215

**Pull request merged**